renderer.hpp

```diff
@@ -567,9 +567,7 @@ void renderer_cool::display_map()
         map_cache->trash();
 
         //If the map size has changed, get rid of it entirely.
-        unsigned int x_bmax, y_bmax, z_max;
-        Maps::getSize(x_bmax, y_bmax, z_max);
-        if (map_cache->maxBlockX() != x_bmax || map_cache->maxBlockY() != y_bmax || map_cache->maxZ() != z_max)
+        if (map_cache->maxBlockX() != world->map.x_count_block || map_cache->maxBlockY() != world->map.y_count_block || map_cache->maxZ() != world->map.z_count_block)
             delete(map_cache);
```

Comment:
Why was this crashing, exactly? It looks to me like Maps::getSize() does what you're doing here, and also performs some validity checks.

Reply:
I have no idea why exactly it was crashing, probably something to do with threads. I just know that the function was crashing sometimes when being called, and when I replaced it with direct reads, it was fine.

I can only assume that the validity check can read a wrong area in memory at some specific points in the loading process.

Reply:
That seems pretty unlikely:
```cpp
bool Maps::IsValid ()
{
    return (world->map.block_index != NULL);
}

// getter for map size
void Maps::getSize (uint32_t& x, uint32_t& y, uint32_t& z)
{
    if (!IsValid())
    {
        x = y = z = 0;
        return;
    }
    x = world->map.x_count_block;
    y = world->map.y_count_block;
    z = world->map.z_count_block;
}

```
I'd say the more likely issue is something trying to use map_cache after it's been deleted.

Reply:
Yeah, crashes were will happening, but not on my machine.

I don't think it's from mapcache, because everything that uses it first checks if it exists, and creates it if it doesn't.

Reply:
I wasn't setting the pointer back to null after deleting the out-dated mapcache. Hopefully that's what was causing the crashes.